src/core/Akka.Persistence.Tests/Serialization/MessageSerializerRemotingSpec.cs

```diff
@@ -202,44 +201,5 @@ public void MessageSerializer_should_custom_serialize_AtomicWrite_messages_durin
                 ExpectMsg("p.b.");
             });
         }
-
-        [Fact]
-        public void MessageSerializer_should_serialize_manifest_provided_by_EventAdapter()
-        {
-            var p1 = new Persistent(new MyPayload("a"), sender: TestActor).WithManifest("manifest");
-            var serializer = _serialization.FindSerializerFor(p1);
-            var bytes = serializer.ToBinary(p1);
-            var back = (Persistent)serializer.FromBinary(bytes, typeof (Persistent));
-
-            back.Manifest.ShouldBe(p1.Manifest);
-        }
-
-        [Fact]
-        public void MessageSerializer_should_serialize_state_change_event()
-        {
-            var p1 = new Persistent(new PersistentFSM.StateChangeEvent("a", TimeSpan.FromSeconds(10)), sender: TestActor);
-            var serializer = _serialization.FindSerializerFor(p1);
-            var bytes = serializer.ToBinary(p1);
-            var back = (Persistent)serializer.FromBinary(bytes, typeof (Persistent));
-            var payload = back.Payload as PersistentFSM.StateChangeEvent;
-            payload.ShouldNotBe(null);
-            payload.StateIdentifier.ShouldBe("a");
-            payload.Timeout.ShouldBeEquivalentTo(TimeSpan.FromSeconds(10));
-        }
-
-        [Fact]
-        public void MessageSerializer_should_serialize_fsm_snapshot()
-        {
-            var snapshot = new PersistentFSM.PersistentFSMSnapshot<MyPayload>("a", new MyPayload("b"), TimeSpan.FromSeconds(10));
-            var p1 = new Persistent(snapshot, sender: TestActor);
```

Comment:
PersistentFSMSnapshot should not be put into `Persistent` object

Reply:
Correct. There is a special case scenario in the `PersistenceMessageSerializer` to handle it. Putting it in the Persistent envelope is wrong.